front/src/modules/ui/table/hooks/useTableColumns.ts

```diff
@@ -36,5 +36,83 @@ export const useTableColumns = () => {
     [tableColumnsByKey, tableColumns, setTableColumns],
   );
 
-  return { handleColumnVisibilityChange };
+  const handleColumnMoveLeftChange = useCallback(
+    (column: ColumnDefinition<ViewFieldMetadata>) => {
+      const tableColumnIndex = tableColumns.findIndex(
+        (tableColumn) => tableColumn.key === column.key,
+      );
+      if (tableColumnIndex >= 0) {
+        const lastColumn = tableColumns[tableColumnIndex - 1];
+        const nextToLastColumn = tableColumns[tableColumnIndex - 2];
+        if (nextToLastColumn) {
+          const newIndex = (lastColumn.index + nextToLastColumn.index) / 2;
+          const updatedColumns = tableColumns
+            .map((previousColumn) =>
+              previousColumn.key === column.key
+                ? { ...previousColumn, index: newIndex }
+                : previousColumn,
+            )
+            .sort((columnA, columnB) => columnA.index - columnB.index);
+
+          setTableColumns(updatedColumns);
+        } else {
+          const nextColumn = tableColumns[tableColumnIndex + 1];
+          const newIndex = (column.index + nextColumn.index) / 2;
+          const updatedColumns = tableColumns
+            .map((previousColumn) =>
+              previousColumn.key === lastColumn.key
+                ? { ...previousColumn, index: newIndex }
+                : previousColumn,
+            )
```

Comment:
The `if` part of logic handles the use-case where the column is moved between other columns. The `else` part handles the edge case where  the current column is to be positioned at the very start. The `nextToLastColumn` will be undefined because there won't be an index present before `lastColumn` index, a mean cannot be calculated in this scenario. So, instead I calculated mean for the current column and `nextColumn` and assigned the value as `lastColumn` index